webui/src/app/item-view/item-view.component.ts

```diff
@@ -56,7 +56,7 @@ export class ItemViewComponent implements OnInit {
   }
 
   handleDelete(): void {
-    if (this.item) {
+    if (this.item && this.item.id) {
```

Comment:
I'm surprised this isn't a linter error -- shouldn't this be written as
```typescript
if (this.item !== null && this.item.id !== null)
```

Reply:
Why that would be better? `this.item` will include `this.item !== null` and `this.item !== undefined`. Though the type of it is `Item | null`, but it can also be changed to `Item | undefined` and the rest will work the same.